scripts/sct_dmri_moco.py

```diff
@@ -40,78 +41,106 @@ def get_parser():
+    parser = argparse.ArgumentParser(
+        description=(
+            '  Motion correction of dMRI data. Some of the features to improve robustness were proposed in Xu et al. '
+            '(http://dx.doi.org/10.1016/j.neuroimage.2012.11.014) and include:\n'
+            '    -group-wise (-g)\n'
+            '    -slice-wise regularized along z using polynomial function (-param). For more info about the method, '
+            'type: isct_antsSliceRegularizedRegistration\n'
+            '    -masking (-m)\n'
+            '    -iterative averaging of target volume\n'
+        ),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Diffusion data. Example: dmri.nii.gz"
+    )
+    mandatory.add_argument(
+        '-bvec',
+        metavar=Metavar.file,
+        required=True,
+        help='Bvecs file. Example: bvecs.txt'
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-bval',
+        metavar=Metavar.file,
+        help='Bvals file. Example: bvals.nii.gz',
+    )
+    optional.add_argument(
+        '-bvalmin',
+        type=float,
+        metavar=Metavar.float,
+        help='B-value threshold (in s/mm2) below which data is considered as b=0. Example: 50.0',
+    )
+    optional.add_argument(
+        '-g',
+        type=int,
+        metavar=Metavar.int,
+        help='Group nvols successive dMRI volumes for more robustness. Example: 2',
+        default=param_default.group_size,
+    )
+    optional.add_argument(
+        '-m',
+        metavar=Metavar.file,
+        help='Binary mask to limit voxels considered by the registration metric. Example: dmri_mask.nii.gz',
+    )
+    optional.add_argument(
+        '-param',
+        metavar=Metavar.str,
+        nargs='+',
+        help=f"R|Advanced parameters. Assign value with \"=\", and separate arguments with a space (\" \")\n"
+             f"    -poly [int]: Degree of polynomial function used for regularization along Z. For no regularization set to "
+             f"0. Default={param_default.poly}.\n"
+             f"    -smooth [mm]: Smoothing kernel. Default={param_default.smooth}.\n"
+             f"    -metric {{MI, MeanSquares, CC}}: Metric used for registration. Default={param_default.metric}.\n"
+             f"    -gradStep [float]: Searching step used by registration algorithm. The higher the more deformation "
+             f"allowed. Default={param_default.gradStep}.\n"
+             f"    -sample [None or 0-1]: Sampling rate used for registration metric. Default={param_default.sampling}.\n"
+    )
```

Comment:
Grouping "advanced parameters" into a single parameter feels non-intuitive and easy to mess up from a user perspective. Grouping in this way also denies us the usual typing and validation that argparse provides. I was thinking that this could instead be an argument group (see "OPTIONAL ARGUMENTS" and "MANDATORY ARGUMENTS" also used here.) 

I left it for now, but this could be changed for all the scripts once the argparse conversion is done.

Reply:
💯 agreed

same comment as above about the cross-compatibility issue and the strategy re: new release

Reply:
Same as above :)